File: sdk/python/apibrew/apiexception.py
class ApiException(Exception):
    def __init__(self, code, message):
        self.code = code
        self.message = message

    def to_dict(self):
        return {'code': self.code, 'message': self.message}

    def __str__(self):
        return f"ApiException: {self.code} {self.message}"

File: sdk/nodejs/src/api-exception.ts
import {Code, Error as ApiBrewError, ErrorField} from "./model/extension";

export class ApiException extends Error {
    constructor(public code: Code = Code.INTERNAL_ERROR, message?: string, public fields?: ErrorField[]) {
        super(message);
    }

    public static fromApiBrewError(error: ApiBrewError): ApiException {
        return new ApiException(error.code, error.message!, error.fields);
    }

    public static fromError(error: any): ApiException {
        if (error.code) {
            return ApiException.fromApiBrewError(error)
        } else {
            return new ApiException(Code.UNKNOWN_ERROR, error.message);
        }
    }

}